Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/CruiseControl.cs

```diff
@@ -0,0 +1,2277 @@
+            else
+                Simulator.Confirmer.Message(ConfirmLevel.Information, Simulator.Catalog.GetString("Selected speed changed to ") + Math.Round(MpS.FromMpS(SelectedSpeedMpS, true), 0, MidpointRounding.AwayFromZero).ToString() + " km/h");
+        }
+
+        public void SpeedRegulatorSelectedSpeedChangeByMouse(float movExtension, bool metric, float maxValue)
+        {
+            if (movExtension != 0 && SelectedMaxAccelerationStep == 0 && DisableCruiseControlOnThrottleAndZeroForce && ForceRegulatorAutoWhenNonZeroSpeedSelectedAndThrottleAtZero &&
+            Locomotive.ThrottleController.CurrentValue == 0 && Locomotive.DynamicBrakeController.CurrentValue == 0 && SpeedRegMode == SpeedRegulatorMode.Manual)
+                SpeedRegMode = SpeedRegulatorMode.Auto;
+            if (movExtension != 0)
+            {
+                if (SelectedSpeedMpS == 0)
+                {
+                    if (movExtension > 0)
+                    {
+                        Locomotive.SignalEvent(Common.Event.LeverFromZero);
+                    }
+                    else if (movExtension < 0)
+                        return;
+                }
+                var deltaSpeed = SpeedSelectorIsDiscrete ? (metric ? MpS.FromKpH((float)Math.Round(movExtension * maxValue / SpeedRegulatorNominalSpeedStepKpHOrMpH) * SpeedRegulatorNominalSpeedStepKpHOrMpH) :
+                    MpS.FromMpH((float)Math.Round(movExtension * maxValue / SpeedRegulatorNominalSpeedStepKpHOrMpH) * SpeedRegulatorNominalSpeedStepKpHOrMpH)) :
+                    (metric ? MpS.FromKpH((float)Math.Round(movExtension * maxValue)) :
+                    MpS.FromMpH((float)Math.Round(movExtension * maxValue)));
+                if (deltaSpeed > 0)
+                    SelectedSpeedMpS = Math.Max(SelectedSpeedMpS + deltaSpeed, MinimumSpeedForCCEffectMpS);
+                else
+                {
+                    SelectedSpeedMpS += deltaSpeed;
+                    if (MinimumSpeedForCCEffectMpS > 0 && SelectedSpeedMpS < MinimumSpeedForCCEffectMpS)
+                        SelectedSpeedMpS = 0;
+                }
+
+                if (SelectedSpeedMpS > Locomotive.MaxSpeedMpS)
+                    SelectedSpeedMpS = Locomotive.MaxSpeedMpS;
+                if (SelectedSpeedMpS < 0)
+                    SelectedSpeedMpS = 0;
+                if (SelectedSpeedMpS == 0 && movExtension < 0)
+                {
+                    Locomotive.SignalEvent(Common.Event.LeverToZero);
+                }
+                if (SpeedIsMph)
+                    Simulator.Confirmer.Message(ConfirmLevel.Information, Simulator.Catalog.GetString("Selected speed changed to ") + Math.Round(MpS.FromMpS(SelectedSpeedMpS, false), 0, MidpointRounding.AwayFromZero).ToString() + " mph");
```

Comment:
Instead of this, could something like Confirm(CabControl.SelectedSpeed, speed) be used?

Reply:
Done as preceding cases